test/stdlib/TestData.swift

```diff
@@ -1183,6 +1183,57 @@ class TestData : TestDataSuper {
         expectEqual(Data(bytes: [1]), slice)
     }
 
+    // This test uses `repeatElement` to produce a sequence -- the produced sequence reports its actual count as its `.underestimatedCount`.
```

Comment:
These tests are all correct, but I find the repeated elements are making it harder to follow the full story. Might I suggest updating this to use `Range`s (which is collection and therefore has exact `count`) and its `Iterator` (which gets upcast to `Sequence`)? 

````swift
let exact = UInt8(0)...UInt8(255)
exact.underestimatedCount // 256
let unknown = exact.makeIterator()
unknown.underestimatedCount // 0
````